### PR TITLE
Add support for `reverse-i-search`

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: src/js/repl.js [hunk 1]
```diff
@@ -133,22 +135,31 @@ export function processLine(replSession: REPLSession, line: string): void {
...
+function stopReverseSearch(
+  replSession: REPLSession,
+  clear: boolean = true,
+): void {
+  const session = replSession;
+  session.isReverseSearch = false;
+  session.reverseSearchBuffer = '';
+
+  if (clear) {
+    session.rl.write(null, {
```
Comment (pesterhazy) [inline src/js/repl.js]: Wouldn't `readline.clearLine` from the api also work?
 https://nodejs.org/api/readline.html#readline_readline_clearline_stream_dir

**Comment by pesterhazy** [comment]
I love it! Comparing with `rlwrap`, it works almost identically (especially love that ^G works as expected). Assuming that the goal is to imitate GNU readline/rlwrap:

- When `rlwrap` runs out of completions, it shows `(failed reverse-i-search)`alpha': alpha`
- Cursor keys (left and right) should cancel reverse-i-search mode. ^F and ^B already work as expected

I had one inline comment, but otherwise this looks good to me. Actually it occurred to me this might be a great general-purpose node library or enhancement to https://www.npmjs.com/package/historic-readline (not saying you should put in the additional work of course).

**Comment by arichiardi** [comment]
I was thinking exactly that when talking about upgradable general purpose node repl. Antonio, you are really raise the bar for node repls here :wink:

File: src/js/repl.js (hunk 1 above)
Comment (anmonteiro) [inline src/js/repl.js]: I did try that and it didn't seem to work for some reason.

**Comment by anmonteiro** [comment]
Thanks for your feedback! I'll definitely incorporate those features. I tried to follow rlwrap style reverse-i-search as close as possible but apparently missed some small behaviors :-)

**Comment by anmonteiro** [comment]
@pesterhazy just added a commit that addressed your early feedback. Mind giving it another go to see if it satisfies all requirements now?

**Comment by pesterhazy** [comment]
@anmonteiro just compiled the updated version, works fabulously. It's hard to tell the difference to "real" readline. 🥇

**Comment by anmonteiro** [comment]
@pesterhazy thanks so much for trying it out. Just did some cleanup wrt. duplicates & entering multiline forms.

Planning to ship it tomorrow morning if nothing major comes up.

**Pull request merged**